### PR TITLE
Makeshift Laser Pistol

### PR DESCRIPTION
**Pull request by assortedbeads**
## About The Pull Request
Adds a laser pistol in the style of the makeshift laser carbine.
Has less damage, similar charge use, steals the cog's double shot 

Raider Pistol 2 
Raider Pistol
RaiderPistolDualBack RaiderPistolDualFront RaiderPistolDualSide RaiderPistolDualTable

## Why It's Good For The Game
Cheaper than the carbine to make and easier to handle, good to pair with a big shield.

## Changelog
:cl:
add: Adds a makeshift laser pistol
add: Adds a makeshift laser pistol recipe
imageadd: added some icons and images for the pistol
/:cl:

### CODE REVIEW

**Comment by Rexdyer1** [comment]
This seems to be a bit too strong damage wise. 1.1 damage multiplier kinda puts it over other laser pistols despite the charge cost you added to the pistol. and since this can be made with literal stock parts, it will blow any other smaller laser out of the water. its cheaper and surprisingly powerful for its size. I'd suggest maybe going for 0.7 or 0.6 with a 1.25 charge cost since you can make this with 200 credits and a trip to the autolathe.

**Comment by Rexdyer1** [comment]
Also correct me if I am wrong, but you put a weapon force instead of a projectile_type? I am not very good at figuring out all the code but after looking at the one star cog code it has a projectile_type and not a weapon force. just thought I'd let you know.

**Comment by DiaFRAME444** [comment]
That's not how it works. All energy weapons need a projectile type so they know what the hell they're firing.
Weapon force is its melee damage.
If a projectile type isn't presented in the item, search its parent types. This item is "/obj/item/gun/energy/laser/makeshift_pistol", so you'd search "/obj/item/gun/energy/laser" for a projectile type, and then "/obj/item/gun/energy" if that doesn't have one either.

To shorten the search, this weapon fires /obj/item/projectile/beam/midlaser, the same as the Lightfall proper. Of course, the lightfall modifies it so it has "damage_multiplier = 1.3", while this has a damage multiplier of 1.1.

**Comment by DiaFRAME444** [comment]
Upon further inspection though, this does seem to be better than the Makeshift Laser Carbine thanks to its burst fire and increased damage multiplier.

**Comment by Rexdyer1** [comment]
ah I see. So its all kinda based on the light fall going down using the normal laser damage. would this make this pistol better. also yeah. aside from the ammo thing which seems negligible. it kinda outclasses almost every other pistol despite being made out of junk and stock parts. you could arm an entire department with these near-round start. other pistols like the svalinn and the martin cost plenty of plasteel to make.

**Comment by assortedbeads** [comment]
> Upon further inspection though, this does seem to be better than the Makeshift Laser Carbine thanks to its burst fire and increased damage multiplier.

Increased? It's got a 1.1 to the makeshift rifles 1.3

**Comment by assortedbeads** [comment]
> ah I see. So its all kinda based on the light fall going down using the normal laser damage. would this make this pistol better. also yeah. aside from the ammo thing which seems negligible. it kinda outclasses almost every other pistol despite being made out of junk and stock parts. you could arm an entire department with these near-round start. other pistols like the svalinn and the martin cost plenty of plasteel to make.

I will just note that while you could arm people with this, you could the same with the rifle version very easily.
It also doesnt have a charge or stun like the martin or spider rose, and has just a 1.1 to their 1
The svalinn is meant to be purposefully bad.

But yeah it might be undernerfed. I dont want to make it hit weaker though since it's meant to be a heavy pistol.
But, I can see it still performing well with a fire delay of 15 instead of ten and more charge cost? So while you get more damage with an opening shot, your consecutive shots are stalled.

**Comment by Rexdyer1** [comment]
yeah. I suppose a higher charge and slower shot would be reasonable, maybe less accuracy? I think the crafting is the biggest issue though. Honestly, when playing vagabond I'd be ecstatic to make a pistol this easily.  however it will heavily change the game as anyone can make a decent laser pistol for dirt cheap.... let's see. what kind of part is not super easy to get but is still doable....maybe having a laser scalpel be part of the process? you could always just add the crystal too, but that usually involves bugging moebius and that's a maybe if they want to help you with that...it needs one part that you can't just buy from a vendor super easily.

**Comment by assortedbeads** [comment]
> yeah. I suppose a higher charge and slower shot would be reasonable, maybe less accuracy? I think the crafting is the biggest issue though. Honestly, when playing vagabond I'd be ecstatic to make a pistol this easily.  however it will heavily change the game as anyone can make a decent laser pistol for dirt cheap.... let's see. what kind of part is not super easy to get but is still doable....maybe having a laser scalpel be part of the process? you could always just add the crystal too, but that usually involves bugging moebius and that's a maybe if they want to help you with that...it needs one part that you can't just buy from a vendor super easily.

It's as easy to make as a makeshift laser carbine, it just has less steel and plastic

**Comment by Rexdyer1** [comment]
nevermind then. I could have sworn the laser carbine used to require a ansible crystal. must have just misremembered. if that's the case then it looks good to me on the crafting front.

**Comment by assortedbeads** [comment]
Yeah you're thinking of the lasblender

**Comment by assortedbeads** [comment]
> yeah that must have been it...say what do you use for your spriting? I've been thinking of adding a couple of junk guns. Worse than the normal guns, but still....decent. 

Just the default thing that pops up when you open a sprite from the game files, forgot what it's called but it's byond stuff

**Comment by Rexdyer1** [comment]
ah so you don't use any special software. sounds easy enough. thanks

**Comment by garrulousTeamotei** [comment]
Going purely off of themes, energy pistols use small cells. 
Dropping charge cost to 50, and having it use small cells could help with some of its balancing.

**Comment by Rexdyer1** [comment]
if you are gonna make it a small cell. you could probably bring the charge to 75-80 and keep the damage.

**Comment by garrulousTeamotei** [comment]
Wait, isn't this just a better version of the sawn off Cog?
An energy gun taking medium cells that can fit in a holster with a smg recoil. But 1.1 damage multiple instead of Cog'

**Comment by assortedbeads** [comment]
> if you are gonna make it a small cell. you could probably bring the charge to 75-80 and keep the damage. 

Im not looking to make it a small cell no

**Comment by Rexdyer1** [comment]
seeing all the benefits and how cheap it is to produce. with most lasers, pistols requiring plasteel, and the big version of this has a lot less storage room. I think it'd be best if you gave it some significant downside. it is a makeshift junk gun. yet its manages to be better than all the other laser guns despite it being made from stock parts that are 40 credits each from a vendor. there should be something that gives someone a reason to consider picking another small laser besides this. I'd say a negative armor pen and higher power cost is the way to go...sorry man but the Gun just seems too good for its price.

also, the Cog is a fine gun. People treat lasers like machine guns or rifles where you charge and click on spaceman until dead. Lasers are better maint weaponry. one time I killed two excels at a distance with a cog because I made sure to hide behind windows and keep my distance. if you can stay out of their accuracy range and keep landing hits you can consistently kill people with good guns and armor....by the way, energy protection should be lower for most clothing seeing how powerful guns are. Don't even really have a reason to use ablative

**Comment by assortedbeads** [comment]
> seeing all the benefits and how cheap it is to produce. with most lasers, pistols requiring plasteel, and the big version of this has a lot less storage room. I think it'd be best if you gave it some significant downside. it is a makeshift junk gun. yet its manages to be better than all the other laser guns despite it being made from stock parts that are 40 credits each from a vendor. there should be something that gives someone a reason to consider picking another small laser besides this. I'd say a negative armor pen and higher power cost is the way to go...sorry man but the Gun just seems too good for its price.
> 
> also, the Cog is a fine gun. People treat lasers like machine guns or rifles where you charge and click on spaceman until dead. Lasers are better maint weaponry. one time I killed two excels at a distance with a cog because I made sure to hide behind windows and keep my distance. if you can stay out of their accuracy range and keep landing hits you can consistently kill people with good guns and armor....by the way, energy protection should be lower for most clothing seeing how powerful guns are. Don't even really have a reason to use ablative

As Ive said before, it is the same as the carbine version but with a lower damage mod.
The carbine version takes a handful of more steel and is the same otherwise as the lightfall, so I dont see how this is going to unbalance the game

**Comment by assortedbeads** [comment]
> > seeing all the benefits and how cheap it is to produce. with most lasers, pistols requiring plasteel, and the big version of this has a lot less storage room. I think it'd be best if you gave it some significant downside. it is a makeshift junk gun. yet its manages to be better than all the other laser guns despite it being made from stock parts that are 40 credits each from a vendor. there should be something that gives someone a reason to consider picking another small laser besides this. I'd say a negative armor pen and higher power cost is the way to go...sorry man but the Gun just seems too good for its price.
> > 
> > also, the Cog is a fine gun. People treat lasers like machine guns or rifles where you charge and click on spaceman until dead. Lasers are better maint weaponry. one time I killed two excels at a distance with a cog because I made sure to hide behind windows and keep my distance. if you can stay out of their accuracy range and keep landing hits you can consistently kill people with good guns and armor....by the way, energy protection should be lower for most clothing seeing how powerful guns are. Don't even really have a reason to use ablative
> 
> As Ive said before, it is the same as the carbine version but with a lower damage mod.
> The carbine version takes a handful of more steel and is the same otherwise as the lightfall, so I dont see how this is going to unbalance the game
>It's also the same size as the carbine Im certain

**Comment by SirRichardFrancis** [comment]
TL;DR

Boring and pointless balance talk, really sick of it.
Gun looks cool, but manufactured. That isn't bad, really, could pass as made by some minor corporation/faction.
But then again, wasn't there a gun ban?

**Comment by Rexdyer1** [comment]
really?

**Comment by SirRichardFrancis** [comment]
Go touch grass please.

**Comment by assortedbeads** [comment]
> TL;DR
> 
> Boring and pointless balance talk, really sick of it.
> Gun looks cool, but manufactured. That isn't bad, really, could pass as made by some minor corporation/faction.
> But then again, wasn't there a gun ban?
> 

I got permission from Firefox to code a new one

**Comment by DiaFRAME444** [comment]
> I can see it still performing well with a fire delay of 15 instead of ten and more charge cost? So while you get more damage with an opening shot, your consecutive shots are stalled.

Sounds fair.

**Comment by Humonitarian** [comment]
Gun embargo.

**Comment by SirRichardFrancis** [comment]
> Gun embargo.

One-man embargo. Poor CeUvi, lmao.

**Comment by assortedbeads** [comment]
> Gun embargo.

Check the other comments, got permission

**Comment by Humonitarian** [comment]
Then completely nerf it to the ground, lol.
It should not be better than a martin. Should be quite identical, in fact.

**Comment by Humonitarian** [comment]
BURST_2_BEAM overrides firerate to my knowledge

**Comment by ValoTheValo** [comment]
wouldn't this be better than the martin/svallin

**Comment by assortedbeads** [comment]
Burst doesnt override fire delay.
It's better for an opening round but as I have said before it has a delay, you get two shots off and then you're staring at the gun while someone kills you.

I have tested this.
A martin and sav are both small cells, the former had taser function, the latter doesnt
Both can be dual wielded to kill or disable easily.

This should be compared to the spider rose which is medium celled
A rose has better efficiency, less delay, can be dualed, has tasers to disarm.
It is better than the makeshift gun in everyway but obtainment.

**Comment by assortedbeads** [comment]
> Then completely nerf it to the ground, lol.
> It should not be better than a martin. Should be quite identical, in fact.

Im not making it identical to a small celled gun with greater functionality, easier storage and dual wield capacity

**Comment by Humonitarian** [comment]
So this is a makeshift cog with more burst damage? Does it do less damage / shot than a proper makeshift laser carbine?

**Comment by DiaFRAME444** [comment]
I mean yeah, its bugging me.
Its supposed to be a downscaled Makeshift Laser Carbine, but it seems to have no relation to it stat wise.
Keeping the medium cell makes sense, having a slower firerate makes sense (Smaller gun, more heat buildup?), but making it deal more damage(?) and fire in bursts (??) compared to the original makes it seem like a completely different design.

**Comment by ErisMeleeEnjoyer** [comment]
can you also discuss Voodoo as much as you discuss a literal trash gun
(i want voodoo merged real bad)

**Comment by assortedbeads** [comment]
> So this is a makeshift cog with more burst damage? Does it do less damage / shot than a proper makeshift laser carbine?

Yes, normal makeshift has a 1.3 modifier, 10 delay, 100 cell charge

This has a 1.1, 13 delay and 125 cell charge
It's in the code

**Comment by DiaFRAME444** [comment]
> Yes, normal makeshift has a 1.3 modifier, 10 delay, 100 cell charge

Trolled by inheritance.

**Comment by assortedbeads** [comment]
> > Yes, normal makeshift has a 1.3 modifier, 10 delay, 100 cell charge
> 
> Oh I got trolled by inheritance.

Yeah. The makeshift carbine actively hits as hard as a lightfall, it has for ages.

**Comment by TheShown911** [comment]
This is what no gun ban does to a PR.

**Comment by assortedbeads** [comment]
I asked before making the PR

File: code/modules/projectiles/guns/energy/laser.dm [hunk 1]
```diff
@@ -301,3 +301,27 @@
...
+	icon = 'icons/obj/guns/energy/makeshift_pistol.dmi'
+	icon_state = "makeshiftpistol"
+	item_state = "makeshiftpistol"
+	origin_tech = list(TECH_COMBAT = 2, TECH_MAGNET = 1)
+	matter = list(MATERIAL_STEEL = 10, MATERIAL_PLASTIC = 7)
+	item_charge_meter = TRUE
+	slot_flags = SLOT_BELT|SLOT_HOLSTER
+	w_class = ITEM_SIZE_NORMAL
+	force = WEAPON_FORCE_NORMAL
+	damage_multiplier = 0.6
```
Comment (Pink-Chink) [inline code/modules/projectiles/guns/energy/laser.dm:317]:
```suggestion
	force = WEAPON_FORCE_NORMAL
	projectile_type = /obj/item/projectile/beam
	damage_multiplier = 0.6
```

File: code/modules/projectiles/guns/energy/laser.dm [hunk 2]
```diff
@@ -301,3 +301,28 @@
...
+	icon_state = "makeshiftpistol"
+	item_state = "makeshiftpistol"
+	origin_tech = list(TECH_COMBAT = 2, TECH_MAGNET = 1)
+	matter = list(MATERIAL_STEEL = 10, MATERIAL_PLASTIC = 7)
+	item_charge_meter = TRUE
+	slot_flags = SLOT_BELT|SLOT_HOLSTER
+	w_class = ITEM_SIZE_NORMAL
+	force = WEAPON_FORCE_NORMAL
+	projectile_type = /obj/item/projectile/beam
+	damage_multiplier = 0.6
```
Comment (Pink-Chink) [inline code/modules/projectiles/guns/energy/laser.dm:318]:
```suggestion
	damage_multiplier = 0.5
```
